token/core/zkatdlog/nogh/driver/driver.go

```diff
@@ -122,23 +128,24 @@ func (d *Driver) NewTokenService(sp driver.ServiceProvider, networkID string, ch
 	}
 
 	ip := identity.NewProvider(sigService, view.GetEndpointService(sp), NewEnrollmentIDDeserializer(), deserializerManager)
-	ws := zkatdlog.NewWalletService(
+	deserializerProvider := NewDeserializerProvider(ppm).Deserialize
+	ws := common.NewWalletService(
+		logger,
 		ip,
-		qe,
-		ppm,
-		NewDeserializerProvider().Deserialize,
-		tmsConfig,
+		deserializerProvider,
+		zkatdlog.NewWalletFactory(ip, qe, tmsConfig, deserializerProvider),
 		identity.NewWalletRegistry(roles[driver.OwnerRole], walletDB),
 		identity.NewWalletRegistry(roles[driver.IssuerRole], walletDB),
 		identity.NewWalletRegistry(roles[driver.AuditorRole], walletDB),
+		nil,
 	)
 	service, err := zkatdlog.NewTokenService(
 		ws,
 		ppm,
 		&zkatdlog.VaultTokenLoader{TokenVault: qe},
 		zkatdlog.NewVaultTokenCommitmentLoader(qe, 3, 3*time.Second),
 		ip,
-		NewDeserializerProvider().Deserialize,
+		NewDeserializerProvider(ppm).Deserialize,
```

Comment:
You may want to use the `deserializerProvider` you declared above (I suppose they are thread safe).

Reply:
👍